lib/nodejs/lobby.js

```diff
@@ -44,6 +44,24 @@ app.get( "/", function( req, res ) {
   } );
 } );
 
+app.get( "/map-list", function( req, res ) {
+  var mapDir = "documents/ITDG/maps";
+
+  // Ensure that the map directory exists, then get the list of files in the map directory
+  mkdirp( mapDir ).then( function( err ) {
+    if ( err ) {
+      console.error( err );
+      res.status( 500 ).send( err );
```

Comment:
In Express, unless the request has a specific response to an error, it's better to pass the error on to `next`. Express will stop routing at that point and will show the application's error page.  Development-mode servers will show the stack trace, and production-mode servers won't, for example.